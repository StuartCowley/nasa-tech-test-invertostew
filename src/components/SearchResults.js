import PropTypes from "prop-types";

import "../styles/SearchResults.css";

function SearchResults({ searchResults }) {
  if (!searchResults.length) {
    return <h3 className="search-results">No search results found.</h3>;
  }

  return (
    <>
      <h3 className="search-results">Search Results</h3>
      <div className="search-results__container">
        {searchResults.map((src) => (
          <img
            key={src}
            src={src}
            alt=""
            className="search-results-img"
            data-testid="image"
          />
        ))}
      </div>
    </>
  );
}

SearchResults.defaultProps = {
  searchResults: [],
};

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(PropTypes.string),
};

export default SearchResults;
