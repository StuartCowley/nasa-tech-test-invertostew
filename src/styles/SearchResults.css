.search-results {
  text-align: center;
  margin: 2rem 0;
}

.search-results-img {
  width: 100%;
  margin-bottom: 1rem;
}

@media (min-width: 576px) {
  .search-results__container {
    display: flex;
    flex-wrap: wrap;
    margin: 0 -0.5rem;
  }

  .search-results-img {
    flex: calc(33.333% - 1rem);
    max-width: calc(33.333% - 1rem);
    padding: 0 0.5rem;
    max-height: 300px;
    object-fit: cover;
  }
}
