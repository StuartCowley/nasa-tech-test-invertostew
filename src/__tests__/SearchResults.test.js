import { render, screen } from "@testing-library/react";

import SearchResults from "../components/SearchResults";

describe("Search", () => {
  const validProps = ["#1", "#2", "#3"];

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults searchResults={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the default results message", () => {
    const emptyProps = [];

    render(<SearchResults searchResults={emptyProps} />);

    const input = screen.getByText(/no search results found/i);

    expect(input).toBeInTheDocument();
  });

  it("renders the correct number of images", () => {
    const { getAllByTestId } = render(
      <SearchResults searchResults={validProps} />
    );

    expect(getAllByTestId("image")).toHaveLength(3);
  });
});
