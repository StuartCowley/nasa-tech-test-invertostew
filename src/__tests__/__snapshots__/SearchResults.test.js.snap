// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <h3
    class="search-results"
  >
    Search Results
  </h3>
  <div
    class="search-results__container"
  >
    <img
      alt=""
      class="search-results-img"
      data-testid="image"
      src="#1"
    />
    <img
      alt=""
      class="search-results-img"
      data-testid="image"
      src="#2"
    />
    <img
      alt=""
      class="search-results-img"
      data-testid="image"
      src="#3"
    />
  </div>
</DocumentFragment>
`;
